mm2src/coins/eth/nft_swap_v2/mod.rs

```diff
@@ -539,3 +675,15 @@ fn erc721_transfer_with_data<'a>() -> Result<&'a ethabi::Function, Erc721Functio
         })?;
     Ok(function)
 }
+
+fn validate_state(tx: &SignedEthTx, state: U256, expected_state: u8) -> Result<(), PrepareTxDataError> {
+    if state != U256::from(expected_state) {
+        return Err(PrepareTxDataError::Internal(ERRL!(
+            "Payment {:?} state is not {}, got {}",
+            tx,
+            expected_state,
+            state
+        )));
+    }
+    Ok(())
```

Comment:
maybe `validate_nft_payment_state` is a better name

Reply:
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2129/commits/40ffb653942419305cfa95eeb71441faed3773d4

renamed to `validate_payment_state`, I think we could reuse it in ETH TPU V2 implementation